File: 4.Median_of_Two_Sorted_Arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        num = nums1 + nums2
        num.sort()
        if len(num)%2 != 0:
            return num[len(num)//2]
        else:
            m1 = num[(len(num)-1)//2]
            m2 = num[(len(num)-1)//2 + 1]
            return (float(m1)+float(m2))/2

if __name__ == '__main__':
    s = Solution()
    print(s.findMedianSortedArrays([1,3], [2]))
    print(s.findMedianSortedArrays([1,2], [3,4]))

File: C++/4.Median_of_Two_Sorted_Arrays.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    double findMedianSortedArrays(const std::vector<int>& nums1, const std::vector<int>& nums2) {
        std::vector<int> num = nums1;
        num.insert(num.end(), nums2.begin(), nums2.end());
        std::sort(num.begin(), num.end());
        
        int n = num.size();
        if (n % 2 != 0) {
            return num[n / 2];
        } else {
            int m1 = num[(n - 1) / 2];
            int m2 = num[(n - 1) / 2 + 1];
            return (static_cast<double>(m1) + static_cast<double>(m2)) / 2;
        }
    }
};

int main() {
    Solution s;
    std::cout << s.findMedianSortedArrays({1, 3}, {2}) << std::endl;
    std::cout << s.findMedianSortedArrays({1, 2}, {3, 4}) << std::endl;
    return 0;
}